### PR TITLE
Add recipe for importmagic.el

### PR DESCRIPTION
**Pull request by anachronic**
### Brief summary of what the package does

Fix python imports using [importmagic](https://github.com/alecthomas/importmagic).

### Direct link to the package repository

https://github.com/anachronic/importmagic.el

### Your association with the package

I'm the creator of the package.

### Relevant communications with the upstream package maintainer

None needed

### Checklist

- [x] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [x] I've used [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] I've built and installed the package using the instructions in the [README](https://github.com/melpa/melpa/blob/master/README.md)

Please go easy on me as I'm not very familiar with Emacs Lisp.

### CODE REVIEW

File: recipes/importmagic [hunk 1]
```diff
@@ -0,0 +1,3 @@
+(importmagic :fetcher github
+             :repo "anachronic/importmagic.el"
+             :files ("importmagic.el" "importmagicserver.py"))
```
Comment (syohex) [inline recipes/importmagic]: Use `:defaults` value instead of specifying each `.el` files.

Please read https://github.com/melpa/melpa#recipe-format

**Comment by syohex** [comment]
Please don't use `C-c i` key binding for Emacs lisp key binding convention.

> Don’t define C-c letter as a key in Lisp programs. Sequences consisting of C-c and a letter (either upper or lower case) are reserved for users; they are the only sequences reserved for users, so do not block them. 

https://www.gnu.org/software/emacs/manual/html_node/elisp/Key-Binding-Conventions.html#Key-Binding-Conventions

**Comment by anachronic** [comment]
Done in both sides: project repo and melpa repo.

**Comment by syohex** [comment]
Looks good.

**Comment by purcell** [comment]
`C-c C-l` is still a binding reserved for major modes. You should at least define the keymap at the top level so it can be customised.

**Comment by anachronic** [comment]
@purcell Thanks for merging.

Is it ok if I leave a suggestion for the user and don't provide a keymap for `importmagic-mode`?

**Comment by purcell** [comment]
> Is it ok if I leave a suggestion for the user and don't provide a keymap for importmagic-mode?

Yep, that's probably a good way to do it.

**Pull request merged**